src/DataBuilder.php

```
@@ -754,13 +754,18 @@ protected function scrub($arr, $fields, $replacement = '*')
             return null;
         }
 
-        $scrubber = function (&$val, $key) use ($fields, $replacement, $arr) {
-
+        $scrubber = function (&$val, $key) use ($fields, $replacement, &$scrubber) {
             if (in_array($key, $fields)) {
                 $val = str_repeat($replacement, 8);
             }
+
+            if (is_array($val)) {
+                array_walk($val, $scrubber);
+            }
         };
-        array_walk_recursive($arr, $scrubber);
```

Argument I'm making here is that if a data type not supported by the scrubbing method is passed we should at least throw an exception and make it very obvious.